app/views/teams/_solved_challenges_table.html.haml

```diff
@@ -4,9 +4,14 @@
       %th= t('teams.summary.solved_challenges_table.challenge_header')
       %th= t('teams.summary.solved_challenges_table.points_header')
       %th= t('teams.summary.solved_challenges_table.when_header')
-  %tbody
-    - @team.solved_challenges.each_with_index do |s, i|
+    = render partial: 'teams/challenge_list_tbody', locals: { teams: @team.solved_challenges.slice(0..4), starting_rank: 1 }
```

Comment:
May make sense to store `solved_challenges` in a variable here, since it is used in 2 spots. It can be added to the `teams_controller.rb` show method, `@solved_challenges = @team.solved_challenges` and then used here.